dbms/src/Storages/DeltaMerge/DeltaMergeStore_Statistics.cpp

```diff
@@ -0,0 +1,239 @@
+
+    stat.delta_placed_rate = static_cast<Float64>(total_placed_rows) / stat.total_delta_rows;
+    stat.delta_cache_size = total_delta_cache_size;
+    stat.delta_cache_rate = static_cast<Float64>(total_delta_valid_cache_rows) / stat.total_delta_rows;
+    stat.delta_cache_wasted_rate = static_cast<Float64>(total_delta_cache_rows - total_delta_valid_cache_rows) / total_delta_valid_cache_rows;
+
+    stat.avg_segment_rows = static_cast<Float64>(stat.total_rows) / stat.segment_count;
+    stat.avg_segment_size = static_cast<Float64>(stat.total_size) / stat.segment_count;
+
+    stat.avg_delta_rows = static_cast<Float64>(stat.total_delta_rows) / stat.delta_count;
+    stat.avg_delta_size = static_cast<Float64>(stat.total_delta_size) / stat.delta_count;
+    stat.avg_delta_delete_ranges = static_cast<Float64>(stat.total_delete_ranges) / stat.delta_count;
+
+    stat.avg_stable_rows = static_cast<Float64>(stat.total_stable_rows) / stat.stable_count;
+    stat.avg_stable_size = static_cast<Float64>(stat.total_stable_size) / stat.stable_count;
+
+    stat.avg_pack_count_in_delta = static_cast<Float64>(stat.total_pack_count_in_delta) / stat.delta_count;
+    stat.avg_pack_rows_in_delta = static_cast<Float64>(stat.total_delta_rows) / stat.total_pack_count_in_delta;
+    stat.avg_pack_size_in_delta = static_cast<Float64>(stat.total_delta_size) / stat.total_pack_count_in_delta;
+
+    stat.avg_pack_count_in_stable = static_cast<Float64>(stat.total_pack_count_in_stable) / stat.stable_count;
+    stat.avg_pack_rows_in_stable = static_cast<Float64>(stat.total_stable_rows) / stat.total_pack_count_in_stable;
+    stat.avg_pack_size_in_stable = static_cast<Float64>(stat.total_stable_size) / stat.total_pack_count_in_stable;
+
+    static const String useless_tracing_id("DeltaMergeStore::getStat");
+    {
+        auto snaps_stat = storage_pool->dataReader()->getSnapshotsStat();
+        stat.storage_stable_num_snapshots = snaps_stat.num_snapshots;
+        stat.storage_stable_oldest_snapshot_lifetime = snaps_stat.longest_living_seconds;
+        stat.storage_stable_oldest_snapshot_thread_id = snaps_stat.longest_living_from_thread_id;
+        stat.storage_stable_oldest_snapshot_tracing_id = snaps_stat.longest_living_from_tracing_id;
+        PageStorage::SnapshotPtr stable_snapshot = storage_pool->dataReader()->getSnapshot(useless_tracing_id);
+        const auto * concrete_snap = toConcreteSnapshot(stable_snapshot);
+        if (concrete_snap)
+        {
+            if (const auto * const version = concrete_snap->version(); version != nullptr)
+            {
+                stat.storage_stable_num_pages = version->numPages();
+                stat.storage_stable_num_normal_pages = version->numNormalPages();
+                stat.storage_stable_max_page_id = version->maxId();
+            }
+            else
+            {
+                LOG_FMT_ERROR(log, "Can't get any version from current snapshot, type=data");
```

Comment:
maybe use `LOG_XXX` instead of `LOG_FMT_XXX`, we will remove all `LOG_FMT_XXX` later.